conanfile.py

```diff
@@ -1,28 +1,33 @@
 from conans import ConanFile, tools
-from conans.tools import download
+import os
+
 
 class JsonForModernCppConan(ConanFile):
     name = "jsonformoderncpp"
-    version = "3.0.1"
+    version = "3.1.0"
     description = "JSON for Modern C++ parser and generator from https://github.com/nlohmann/json"
     license = "MIT"
     url = "https://github.com/vthiery/conan-jsonformoderncpp"
+    repo_url = "https://github.com/nlohmann/json"
     author = "Vincent Thiery (vjmthiery@gmail.com)"
+    options = {"single_header": [True, False]}
+    default_options = "single_header=True"
 
     def source(self):
-        download("https://github.com/nlohmann/json/releases/download/v%s/json.hpp" % self.version, "json.hpp")
+        release_url = "%s/releases/download/v%s" % (self.repo_url, self.version)
+
+        tools.download("%s/blob/v%s/LICENSE.MIT" % (self.repo_url, self.version), "LICENSE.MIT")
 
-    def build(self):
-        # as there is no LICENSE file, lets extract and generate it.
-        # It is useful for package consumers, so they can collect all license from all dependencies
-        tmp = tools.load("json.hpp")
-        license_contents = tmp[2:tmp.find("*/", 1)]
-        tools.save("LICENSE", license_contents)
+        if self.options.single_header:
```

Comment:
Using options in ``source()`` doesn't work that way. If you try to use the other option, it will fail.

``source()`` must be common for all settings and options. If anything I would move that to ``build()``

Reply:
Right, my bad.